src/audio/buffers/comp_buffer.c

```diff
@@ -480,9 +480,10 @@ void comp_update_buffer_produce(struct comp_buffer *buffer, uint32_t bytes)
 	/* return if no bytes */
 	if (!bytes) {
 #if CONFIG_SOF_LOG_DBG_BUFFER
+		struct comp_dev *src_component = comp_buffer_get_source_component(buffer);
```

Comment:
This would need a "__unused" attribute to not create compiler warnings when buf_dbg() is a no-op.